File: Medias.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AutomatizarPruebasUnitarias {
    
    class Medias {

        /**
         * Calcula y regresa la media artimética
         */
        public static double mediaAritmetica(params int[] vals) { }
            
        class promedio{
            private double[] valores = new doble[vals.Lenght];
            private int n = 0;
            
            public void llenar(double valor)
            {
                if(n== vals.Lenght){}
                else{
                    valores[n] = valor;
                }
            }
            public double promedio(){
             double prom = 0;
             double suma = 0;
             for (int i = 1; i < n; i++){
                 suma = suma + valores[i];
             }
             prom = suma / n;
             return prom;
         }
        }
        /**
         * Calcula y regresa la raíz enésima = x^(1/n)
         */
        private static double raizEnesima(double x, int n) { }
            double raizEnesima = x^(1/n);
            return raizEnesima;
        /**
         *  Usa raizEnesima para calcular y regresar la media geométrica
         */
        public double mediaGeometrica(params int[] vals) { }
        double mediaAritmetica = 0;
        double producto = 0;
        for (int i = 0; i < vals.Lenght[]; i++){
            producto = producto*vals[i];
        }
        return raizEnesima;
            
        /**
         * Este método no está implementado.
         */
        public static double mediaArmonica(params int[] vals) { }
    }
}

File: Medias.py
import math


class Medias:
    """Metodo que obtiene la media geometrica. params: double vals"""
    def mediaAritmetica(self, vals):
        return sum(vals) / len(vals)

    """Metodo que la raiz enesima. params: double num, int k"""
    def raizEnesima(self, num, k):
        return math.pow(num, (1.0 / k))

    def mediaGeometrica(self, vals):
        prod = 1
        for n in vals:
            prod *= n
        return self.raizEnesima(prod, len(vals))

    def mediaArmonica(self, vals):
        sum = 0
        for n in vals:
            sum += 1 / n
        return len(vals) / sum


def main():
    medias = Medias()
    print(medias.mediaAritmetica([2.0, 4.0, 8.0]))
    print(medias.raizEnesima(8.0, 3.0))
    print(medias.mediaGeometrica([1.0, 2.0, 4.0]))
    print(medias.mediaArmonica([2.0, 4.0, 8.0]))


if __name__ == "__main__":
    main()
